src/genTypeFlow.ts

```diff
@@ -36,18 +36,38 @@ function getResourceArg(resourceConfig: ResourceConfig, resourcePath: ReadonlyAr
         >`;
 }
 
+export function getNewKeyTypeFromBatchKeySetType(batchKey: string, resourceArgs: string) {
+    return `\
+        $Call<
+            ExtractArg,
+            [$PropertyType<$PropertyType<${resourceArgs}, '${batchKey}'>, 'has'>]
```

Comment:
doesn't `has` return a boolean rather than the underlying value?

https://github.com/facebook/flow/blob/388ea046bb7ca6f64fda17637daae8a3440ae0b2/lib/core.js#L600

slightly confused here sorry!

Reply:
Yeah, finding a solution here wasn't easy/obvious, even when I was pairing on it with @ATRAN2. This is extracting the arg type of `has()` (`T`) similarly to how the code extracts the batchKey arg type to the batch resource function (ie `Set<T>` in this case) itself.

Reply:
ohhhh this is the type of the newKey, sorry i'm being dumb

looks good, nice approach :) 👍